lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufColumnDecoder.java

```diff
@@ -144,4 +177,36 @@ private static Optional<Descriptor> getDescriptor(FieldDescriptor fieldDescripto
         }
         return Optional.empty();
     }
+
+    private static Object createOneofJson(DynamicMessage message, Set<String> oneofColumns)
```

Comment:
Any specific reason on why we should pass the `oneofColumns` from the KafkaColumnHandle ? Why can't we use `message.getDescriptorForType()` get oneOf types from here and handle them ? In this case we don't have to change other decoder right ?

Reply:
Yeah. You can collect `oneof` fields from `FieldDescriptor`:

`.getContainingOneof().getName()`

Reply:
Great, thanks for the suggestion. I just couldn't find a way to get the names to where they needed to be, but this does the trick. This removes the need to include the `oneof` names in the column handles.